ttnn/cpp/ttnn/operations/eltwise/complex/complex.hpp

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <functional>
+#include <optional>
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/core/core.hpp"
+
+namespace ttnn {
+namespace operations::complex {
+
+class ComplexTensor {
+    private:
+        std::array<Tensor,2> m_real_imag;
+
+    public:
+
+        ComplexTensor(std::array<Tensor,2> val): m_real_imag(val) {
+            TT_ASSERT( m_real_imag[0].get_legacy_shape() == m_real_imag[1].get_legacy_shape() , "Tensor shapes of real and imag should be identical");
+        }
+
+        const Tensor& operator[](uint32_t index) const {
+            return m_real_imag[index];
+        }
+
+        const Tensor& real() const {
+            return m_real_imag[0];
+        }
+
+        const Tensor& imag() const {
+            return m_real_imag[1];
+        }
+
+        void deallocate() {
+            m_real_imag[0].deallocate();
+            m_real_imag[1].deallocate();
+        }
+};
+
+struct ExecuteGetComplexTensor {
+
+    static ComplexTensor operator()(
+        const Tensor &input_tensor_a_arg,
+        const Tensor &input_tensor_b_arg,
+        const std::optional<MemoryConfig> &memory_config) {
+        return ComplexTensor({input_tensor_a_arg, input_tensor_b_arg});
```

Comment:
Lets move implementation to cpp too. It is not heavy, but will be consistent with other things.

Reply:
done